secret.go

```diff
@@ -0,0 +1,53 @@
+package pipeline
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	_ interface {
+		json.Marshaler
+		yaml.Marshaler
+		selfInterpolater
+	} = (*Secret)(nil)
+)
+
+// Secret represents a pipeline secret configuration.
+type Secret struct {
+	Key                 string         `json:"key" yaml:"key"`
+	EnvironmentVariable *string        `json:"environment_variable,omitempty" yaml:"environment_variable,omitempty"`
+	RemainingFields     map[string]any `yaml:",inline"`
+}
+
+// MarshalJSON marshals the secret to JSON.
+func (s *Secret) MarshalJSON() ([]byte, error) {
+	type secretAlias Secret
+	return json.Marshal((*secretAlias)(s))
+}
```

Comment:
This pretty much implements the default behaviour. 
The main reason to provide a custom `MarshalJSON` implementation here would be to support the `"inline"`-tagged field (`RemainingFields`), which we have a pre-written function for:


```suggestion
// MarshalJSON marshals the secret to JSON.
func (s *Secret) MarshalJSON() ([]byte, error) {
	return inlineFriendlyMarshalJSON(s)
}
```

(`"inline"` is a `yaml`-package convention, but not one supported by the stdlib `encoding/json`).

If we don't need to future-proof the struct (i.e. we declare right now that `key` and `environment_variable` are the only attributes, or that if we decide to support them, they require e.g. an agent upgrade), then we don't need `RemainingFields` or this custom marshaler method at all.